nginx/1.25.3-zlib-ng.patch

```diff
@@ -0,0 +1,341 @@
++
+         /*
+          * The zlib deflate_state allocation, it takes about 6K,
+          * we allocate 8K.  Other allocations are divisible by 512.
+          */
+ 
+-        ctx->state_allocated = 1;
+-
+         alloc = 8192;
+     }
+ 
+@@ -954,13 +957,13 @@
+         return p;
+     }
+ 
+-    if (ctx->zlib_ng) {
++    if (ctx->intel) {
+         ngx_log_error(NGX_LOG_ALERT, ctx->request->connection->log, 0,
+                       "gzip filter failed to use preallocated memory: "
+                       "%ud of %ui", items * size, ctx->allocated);
+ 
+     } else {
+-        ngx_http_gzip_assume_zlib_ng = 1;
++        ngx_http_gzip_assume_intel = 1;
+     }
+ 
+     p = ngx_palloc(ctx->request->pool, items * size);
+diff -urN nginx-1.25.3.orig/src/http/modules/ngx_http_log_module.c nginx-1.25.3/src/http/modules/ngx_http_log_module.c
+--- nginx-1.25.3.orig/src/http/modules/ngx_http_log_module.c	2023-10-24 15:46:47.000000000 +0200
++++ nginx-1.25.3/src/http/modules/ngx_http_log_module.c	2023-11-25 04:00:16.021729227 +0100
+@@ -9,8 +9,13 @@
+ #include <ngx_core.h>
+ #include <ngx_http.h>
+ 
+-#if (NGX_ZLIB)
+-#include <zlib.h>
++#if defined(NGX_ZLIB_NG)
++# include <zlib-ng.h>
++# define ZPREFIX(x) zng_ ## x
++# define z_stream zng_stream
++#elif defined(NGX_ZLIB)
++# include <zlib.h>
++# define ZPREFIX(x) x
+ #endif
+ 
+ 
+@@ -634,7 +639,7 @@
+     zstream.next_out = out;
+     zstream.avail_out = size;
+ 
+-    rc = deflateInit2(&zstream, (int) level, Z_DEFLATED, wbits + 16, memlevel,
++    rc = ZPREFIX(deflateInit2)(&zstream, (int) level, Z_DEFLATED, wbits + 16, memlevel,
+                       Z_DEFAULT_STRATEGY);
+ 
+     if (rc != Z_OK) {
+@@ -647,7 +652,7 @@
+                    zstream.next_in, zstream.next_out,
+                    zstream.avail_in, zstream.avail_out);
+ 
+-    rc = deflate(&zstream, Z_FINISH);
++    rc = ZPREFIX(deflate)(&zstream, Z_FINISH);
+ 
+     if (rc != Z_STREAM_END) {
+         ngx_log_error(NGX_LOG_ALERT, log, 0,
+@@ -663,7 +668,7 @@
+ 
+     size -= zstream.avail_out;
+ 
+-    rc = deflateEnd(&zstream);
++    rc = ZPREFIX(deflateEnd)(&zstream);
+ 
+     if (rc != Z_OK) {
+         ngx_log_error(NGX_LOG_ALERT, log, 0, "deflateEnd() failed: %d", rc);
+diff -urN nginx-1.25.3.orig/src/stream/ngx_stream_log_module.c nginx-1.25.3/src/stream/ngx_stream_log_module.c
+--- nginx-1.25.3.orig/src/stream/ngx_stream_log_module.c	2023-10-24 15:46:47.000000000 +0200
++++ nginx-1.25.3/src/stream/ngx_stream_log_module.c	2023-11-25 04:00:16.021729227 +0100
+@@ -9,8 +9,13 @@
+ #include <ngx_core.h>
+ #include <ngx_stream.h>
+ 
+-#if (NGX_ZLIB)
+-#include <zlib.h>
++#if defined(NGX_ZLIB_NG)
++# include <zlib-ng.h>
++# define ZPREFIX(x) zng_ ## x
++# define z_stream zng_stream
++#elif defined(NGX_ZLIB)
++# include <zlib.h>
++# define ZPREFIX(x) x
+ #endif
+ 
+ 
+@@ -482,7 +487,7 @@
+     u_char      *out;
+     size_t       size;
+     ssize_t      n;
+-    z_stream     zstream;
++    z_stream   zstream;
```

Comment:
This whitespace change seems unnecessary.

Reply:
That change is in the original patch for 1.18, so didn't notice it.

Reply:
fixed